base/linalg/triangular.jl

```diff
@@ -889,6 +889,201 @@ for (f, g) in ((:/, :A_rdiv_B!), (:A_rdiv_Bc, :A_rdiv_Bc!), (:A_rdiv_Bt, :A_rdiv
     end
 end
 
+# Complex matrix logarithm for the upper triangular factor, see:
+#   Al-Mohy and Higham, "Improved inverse  scaling and squaring algorithms for
+# the matrix logarithm", SIAM J. Sci. Comput., 34(4), (2012), pp. C153-C169.
+#   Al-Mohy, Higham and Relton, "Computing the Frechet derivative of the matrix
+# logarithm and estimating the condition number", SIAM J. Sci. Comput., 35(4),
+# (2013), C394-C410.
+function logm{T}(A0::UpperTriangular{T})
+    maxsqrt = 100
+    theta = [1.586970738772063e-005,
+         2.313807884242979e-003,
+         1.938179313533253e-002,
+         6.209171588994762e-002,
+         1.276404810806775e-001,
+         2.060962623452836e-001,
+         2.879093714241194e-001]
+    tmax = size(theta, 1)
+    n = size(A0, 1)
+    A = A0
+    p = 0
+    m = 0
+
+    # Compute repeated roots
+    d = diag(A)
+    dm1 = Array(T, n)
+    s = 0
+    for i = 1:n
+        dm1[i] = dm1[i] - 1.
+    end
+    while norm(dm1, Inf) > theta[tmax]
+        for i = 1:n
+            d[i] = sqrt(d[i])
+        end
+        for i = 1:n
+            dm1[i] = d[i] - 1
+        end
+        s = s + 1
+    end
+    s0 = s
+    for k = 1:min(s, maxsqrt)
+        A = sqrtm(A)
+    end
+
+    AmI = A - UpperTriangular(eye(n))
```

Comment:
You can do this more efficiently with `A - I`. `I` is a special constant of type `UniformScaling`.